### DISCUSSION TITLE
Collection select()

### DISCUSSION THREAD

**Discussion by roberto-butti** [laravel/framework#41521] [Ideas]
- Class: Collection
- Method: select()

When you have a bi dimensional array like this one (an array of associative arrays):
```
    [
            ['product' => 'Desk', 'price' => 200, 'active' => true],
            ['product' => 'Chair', 'price' => 100, 'active' => true],
            ['product' => 'Door', 'price' => 300, 'active' => false],
            ['product' => 'Bookcase', 'price' => 150, 'active' => true],
            ['product' => 'Door', 'price' => 100, 'active' => true],
        ]
```
You can apply method for filtering (**where()**), sorting (**sortBy()**) and so on ...
If you want to select and "pick" just few keys or columns or fields, probably, as a developer, you want to use a method to specify the keys you want:
```
$collection->select(["product", "price"])
        ->where("price", ">", 100)
        ->sortBy("price")
        ->all()
```
I would like to suggest implementing "**select**()" method where the developer can define a list of keys to "extract".
Think to a bi- dimensional array like a db table where you have columns, and you need to perform a select operation to select just some columns.

The result is something like that:
```php
$a = Collection::make(
    [
            ['product' => 'Desk', 'price' => 200, 'active' => true],
            ['product' => 'Chair', 'price' => 100, 'active' => true],
            ['product' => 'Door', 'price' => 300, 'active' => false],
            ['product' => 'Bookcase', 'price' => 150, 'active' => true],
            ['product' => 'Door', 'price' => 100, 'active' => true],
        ]
)
        ->select(["product", "price"])
        ->where("price", ">", 100)
        ->sortBy("price")
        ->all();
var_export($a);
/*
array (
  3 =>
  array (
    'product' => 'Bookcase',
    'price' => 150,
  ),
  0 =>
  array (
    'product' => 'Desk',
    'price' => 200,
  ),
  2 =>
  array (
    'product' => 'Door',
    'price' => 300,
  ),
)
*/
```

I could have the code already implemented. I added also the static method to Arr class.
If you agree with this suggestion, I can open a PR, if not: "amici come prima 🇮🇹 " 🥳

**Comment by mikeydevelops** [comment]
There is already a method that does what you want, although it is called [`only`](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Collections/Collection.php#L862) instead of `select`

**Reply by chuckrincon** [reply]
The `only` function works with one dimensional arrays, not with multi dimensional arrays.

**Reply by mikeydevelops** [reply]
Yeah... sorry about that. Little bit brain farted there :)

**Comment by chuckrincon** [comment]
There was a [discussion](https://github.com/laravel/framework/issues/13667) before about this specific case.
So a PR adding this feature would be nice.

**Comment by sertxudev** [comment]
You can use a combination of `map` and `only`.
For example:
```php
$collection->map->only(['product', 'price']);
```
